doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,260 @@
+	maximum_current         = float                          ; Maximum recorded measurement
+	minimum_current         = float                          ; Mininum recorded measurement
+	
+		
+#### sonic-platform-common
+
+Chassis Base class will be enhanced with prototype methods for retrieving number of sensors and sensor objects of a specific type.
+
+Module base class will also be enhanced with similar methods for retrieving sensors present on the modules.
+
+New base classes will be introduced for new sensor types.
+
+VsensorBase is introduced for voltage sensor objects. 
+IsensorBase is introdued for current sensor objects.
+
+The classes will have methods to retrieve threshold information, sensor value and min/max recorded values from the sensor.
+	
+#### sonic-utilities
+	
+CLIs are introduced to retrieve and display sensor data from State DB for different sensor types. CLIs are described in the next section.
+
+#### sonic-buildimage
+
+The CLI "show system-health" should report sensor fault conditions. Hardware health check script will need enhancement to retrieve sensor data from StateDB. 
+
+#### sonic-snmpagent
+
+Voltage and Current sensors should be available in Entity MIB. Entity MIB implementation will need an enhancement to retrieve voltage and current sensors from the state DB.
+
+### CLI Enhancements 
+
+Following CLI is introduced to display the Voltage and Current Sensor devices.
+
+	root@sonic:/home/cisco# show platform voltage
+	          Sensor       Voltage    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
+	----------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
+	VP0P75_CORE_NPU0         750 mV        852       684             872            664      False  20230204 11:35:21
+	VP0P75_CORE_NPU1         750 mV        852       684             872            664      False  20230204 11:35:21
+	VP0P75_CORE_NPU2         750 mV        852       684             872            664      False  20230204 11:35:22
+	
+	...
+	
+	root@sonic:/home/cisco# show platform current
+	        Sensor        Current    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
+	--------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
+	POL_CORE_N0_I0       25000 mA      30000     18000           28000          15000      False  20230212 11:18:28
+	POL_CORE_N0_I1       21562 mA      30000     18000           28000          15000      False  20230212 11:18:28
+	POL_CORE_N0_I2       22250 mA      30000     18000           28000          15000      False  20230212 11:18:28
+
+
+	
+
+#### Configuration and Management
+
+At this point, there is no configuration requirement for this feature. 
+
+If the SensorMon daemon is not desired to be run in the system, an entry can be added to  pmon_daemon_control.json to exclude it from running in the system.
```

Comment:
@bmridul can we ensure that existing platform don't have to explicitly follow this i.e if sensor.ymal is not present then sensormon does not run?

Reply:
If platform does not provide the sensor definition, then there is no impact. Sensormond will not do anything. The pmon_daemon_control.json is simply if the platform does not want to have the daemon running. This is same approach used for other PMON daemons (e.g. Same approach is for PSUd - https://github.com/sonic-net/sonic-buildimage/blob/927dde73f12629f3a4da84a729fa6ea65cad9676/dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2#L134)